### DISCUSSION TITLE
Disable shorts per channel

### DISCUSSION THREAD

**Discussion by newhinton** [libre-tube/LibreTube#6460] [Ideas]
### Describe your suggested feature

There are some channels that only post short-videos.

There are other channels that spam short-videos.

At the moment, it is only possible to enable or disable shorts in the main view for every subscription.

It would be nice if it were possible to more selectively filter content, so that i can keep the long-form content of the spammy channels, and the shorts from the not spammy ones.

Thank you for your app!

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Livid96** [comment]
I guess 'do not show less then "T" second video" should do the work.

**Comment by newhinton** [comment]
I can't find the setting right now, but last time i checked this applied to all channels. That is not what i want, i want to select that per-channel

**Comment by Livid96** [comment]
I'm not a dev so W8 untill dev so something.
Seems like you are a dev. Can't you try by Your self.

**Comment by Bnyro** [comment]
It's perhaps not difficult to implement, but I'm not really a fan of that idea to allow such per-channel settings.

Previously, blocking channels has already been rejected for a similar reason, the only current per-channel setting is about enabling/disabling notifications for that specific channel.

I'm really wondering if such options are a good idea because
- they add lots of additional code that needs to be maintained
- make the user interface more complicated, especially for new users this can be confusing.

**Comment by newhinton** [comment]
>  the only current per-channel setting is about enabling/disabling notifications for that specific channel.

Please don't remove that :laughing: 

I understand those reasons. I was myself debating on how to do that ux-wise.

Maybe a more general approach would be an option:
In the settings, show a "filter" section. For each filter, a channel can be added, a type and a "filter" (regex-style, but limited to wildcards)
This filter could apply to the title or to the tags, and would hide the element from the subscription-view only, and from notifications.

This would remove it from immediate view for new users, reducing the cluttering of the "main screens", but still allow a widespread control over what is shown. It would cover my usecase, but others aswell.

Though, generally i agree that it should not get overly complicated for maintenance reasons.

**Comment by Bnyro** [comment]
> It would be nice if it were possible to more selectively filter content, so that i can keep the long-form content of the spammy channels, and the shorts from the not spammy ones.
>
In that case it would maybe make sense to add a filter option to only show `Shorts (unspammed)`, which just counts the amount of shorts uploaded per channel and limit it to a fixed number e.g. 3. Shorts of any channel with more than 3 shorts in the feed (past month) will just not be displayed. This will break if we ever add server side feed pagination at Piped though (which is unlikely to happen anytime soon).

I'm neither convinced by my suggestion nor by yours though, I think that such a big filtering system is just an overkill for the app and ugly in terms of maintainability.
